source/_posts/2017/2017-02-15-minimalistic-way-to-create-your-first-nette-extension.md

```diff
@@ -0,0 +1,209 @@
+---
+layout: post
+title: "Minimalistic Way to Create Your First Nette Extension"
+perex: '''
+    Nette extension allows you not only to create open-source packages, but also to <strong>split your application to small and logical chunks of code</strong>.
+    <br><br>
+Open-source extensions are more complex ěusing many Nette\DI features, but today I will show you, how to <strong>start with one Nette\DI method and one service only</strong>.
+'''
+lang: en
+---
+
+First, I will tell you a little trick, **how to make learning faster and easier**. Don't worry, after this one headline we get to the code.
+
+## Set The Smallest Step Possible
+
+When I want to learn new skill, I try to realize: what is **the smallest step possible**? Also called *lean method*, or in software [Lean Software Development](https://en.wikipedia.org/wiki/Lean_software_development) (LSD, right?).
+
+### I Learned Extensions the Hard Way
+
+1. **By Reading Nette documentation** [that describes over 20 features and uses cases it has](https://doc.nette.org/en/last/di-extensions). That is information, which is useful, because I can use the tool to it's potential, but not the best adopt the skill.
+
+2. **By Reading extension of open-source packages**. They are similar to documentation: many features on various use cases I didn't understand yet.
+
+That lead to **overstretching my brain muscle**. It's like trying to jump over huge hole before even walking.
+
+
+### It made me think: "Could it be simpler?"
+
+What is essential purpose of the extension? It registers services to Nette Service Container.
+
+- Register services to Nette Container?
```

Comment:
Tady už se to snažím zjednodušit na co nejméně slov. Je jedno, jestli je to Nette. Je jedno jestli DI. Je to jen nějaká schránka na data.

 Asi nechám jen `Container`